configure.ac

```diff
@@ -2971,15 +2995,30 @@ then
 	SCO_SV*)
 		LDSHARED='$(CC) -Wl,-G,-Bexport'
 		LDCXXSHARED='$(CXX) -Wl,-G,-Bexport';;
+	WASI*)
+		AS_VAR_IF([enable_wasm_dynamic_linking], [yes], [
+			dnl not iplemented yet
+		]);;
 	CYGWIN*)
 		LDSHARED="gcc -shared -Wl,--enable-auto-image-base"
 		LDCXXSHARED="g++ -shared -Wl,--enable-auto-image-base";;
 	*)	LDSHARED="ld";;
 	esac
 fi
+
+dnl Emscripten's emconfigure sets LDSHARED. Set BLDSHARED outside the
+dnl test -z $LDSHARED block to configure BLDSHARED for side module support.
+if test "$enable_wasm_dynamic_linking" = "yes" -a "$ac_sys_system" = "Emscripten"; then
+  BLDSHARED='$(CC) -shared -sSIDE_MODULE=1 -sWASM=1'
```

Comment:
Is there some reason why you add `-sWASM=1` here.   Since its the default its unlikely you need it.

Also, you can simplify this command line (and the ones elsewhere in this file) by dropping with `=1` at the end of all the settings.

Reply:
Python shared extensions have a ``.so`` extension. emcc creates a ``.so`` JavaScript file and a ``.wasm`` file along the ``.so`` file. We want the ``.so`` file to contain the ``.wasm`` code.

Reply:
This is not what the `-sWASM=1` option does though.   The `-sSIDE_MODULE` option is enough to ensure that the output is just a wasm file.

If you are building a normal application without `-sSIDE_MODULE` then you always get a JS file and a wasm file side-by-side (even with `-sWASM=1`).   The `-sWASM=1` option (which is the default) simply means "don't convert the wasm file to JS using wasm2js".. which is what happens if you set `-sWASM=0` (which is never the default).

`-sSIDE_MODULE` on its own should always build just a wasm file.

Reply:
Please note this PR is merged already, so it would be best to open a new issue to track this discussion if there's something to change.